pkg/storage/backend/swift/config.go

```diff
@@ -0,0 +1,46 @@
+package swift
+
+import (
+	"flag"
+)
+
+// Config holds the config options for GCS backend
+type Config struct {
+	AuthUrl           string `yaml:"auth_url"`
+	Username          string `yaml:"username"`
+	UserDomainName    string `yaml:"user_domain_name"`
+	UserDomainID      string `yaml:"user_domain_id"`
+	UserId            string `yaml:"user_id"`
+	Password          string `yaml:"password"`
+	DomainId          string `yaml:"domain_id"`
+	DomainName        string `yaml:"domain_name"`
+	ProjectID         string `yaml:"project_id"`
+	ProjectName       string `yaml:"project_name"`
+	ProjectDomainID   string `yaml:"project_domain_id"`
+	ProjectDomainName string `yaml:"project_domain_name"`
+	RegionName        string `yaml:"region_name"`
+	ContainerName     string `yaml:"container_name"`
+}
+
+// RegisterFlags registers the flags for TSDB Swift storage
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	cfg.RegisterFlagsWithPrefix("blocks-storage.", f)
+}
+
+// RegisterFlagsWithPrefix registers the flags for TSDB Swift storage with the provided prefix
+func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&cfg.AuthUrl, prefix+"swift.auth-url", "", "Openstack authentication URL")
+	f.StringVar(&cfg.Username, prefix+"swift.username", "", "Openstack username for the api.")
+	f.StringVar(&cfg.UserDomainName, prefix+"swift.user-domain-name", "", "Openstack user's domain name.")
+	f.StringVar(&cfg.UserDomainID, prefix+"swift.user-domain-id", "", "Openstack user's domain id.")
+	f.StringVar(&cfg.UserId, prefix+"swift.user-id", "", "Openstack userid for the api.")
```

Comment:
Could you also clarify the difference between "username", "user domain" and "user ID"? Are they mutually exclusive or not? When should be set?

```suggestion
	f.StringVar(&cfg.UserId, prefix+"swift.user-id", "", "OpenStack Switch user ID.")
```

Reply:
That's kinda complicated. Not sure should we put that in cli help. It's about different Swift Auth API versions. See e.g. https://godoc.org/github.com/ncw/swift 
For v1 (which is obsolete) - only URL, username and password are needed.
For V2 - URL, username, password are mandatory, then tenant name OR tenant ID is mandatory, and region is mandatory.
V3 is even more complex, all these user domain/id, project domain/id are related to it.
Usually, Swift users knows exactly which auth version is supported and which parameters needs to be provided.

Reply:
Let me know what's better to do here, please